server/config/config_test.go

```diff
@@ -76,6 +82,259 @@ func TestConfigRoundtrip(t *testing.T) {
+				RedisScanKeysCount:      1000,
+			},
+		},
+		{
+			desc: "yaml collect interval lock timeout",
+			yaml: `
+osquery:
+  enable_async_host_processing: true
+  async_host_collect_interval: 10s
+  async_host_collect_lock_timeout: 20s`,
+			wantLabelCfg: AsyncProcessingConfig{
+				Enabled:                 true,
+				CollectInterval:         10 * time.Second,
+				CollectMaxJitterPercent: 10,
+				CollectLockTimeout:      20 * time.Second,
+				CollectLogStatsInterval: 1 * time.Minute,
+				InsertBatch:             2000,
+				DeleteBatch:             2000,
+				UpdateBatch:             1000,
+				RedisPopCount:           1000,
+				RedisScanKeysCount:      1000,
+			},
+		},
+		{
+			desc: "yaml collect interval lock timeout per task",
+			yaml: `
+osquery:
+  enable_async_host_processing: true
+  async_host_collect_interval: label_membership=10s
+  async_host_collect_lock_timeout: policy_membership=20s`,
+			wantLabelCfg: AsyncProcessingConfig{
+				Enabled:                 true,
+				CollectInterval:         10 * time.Second,
+				CollectMaxJitterPercent: 10,
+				CollectLockTimeout:      1 * time.Minute,
+				CollectLogStatsInterval: 1 * time.Minute,
+				InsertBatch:             2000,
+				DeleteBatch:             2000,
+				UpdateBatch:             1000,
+				RedisPopCount:           1000,
+				RedisScanKeysCount:      1000,
+			},
+		},
+		{
+			desc: "yaml env var override",
+			yaml: `
+osquery:
+  enable_async_host_processing: false
+  async_host_collect_interval: label_membership=10s
+  async_host_collect_lock_timeout: policy_membership=20s
+  async_host_insert_batch: 10`,
+			envVars: []string{
+				"FLEET_OSQUERY_ENABLE_ASYNC_HOST_PROCESSING=policy_membership=false&label_membership=true",
+				"FLEET_OSQUERY_ASYNC_HOST_COLLECT_INTERVAL=policy_membership=30s&label_membership=50s",
+				"FLEET_OSQUERY_ASYNC_HOST_COLLECT_LOCK_TIMEOUT=40s",
+			},
+			wantLabelCfg: AsyncProcessingConfig{
+				Enabled:                 true,
+				CollectInterval:         50 * time.Second,
+				CollectMaxJitterPercent: 10,
+				CollectLockTimeout:      40 * time.Second,
+				CollectLogStatsInterval: 1 * time.Minute,
+				InsertBatch:             10,
+				DeleteBatch:             2000,
+				UpdateBatch:             1000,
+				RedisPopCount:           1000,
+				RedisScanKeysCount:      1000,
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			var cmd cobra.Command
+			// Leaving this flag unset means that no attempt will be made to load
+			// the config file
+			cmd.PersistentFlags().StringP("config", "c", "", "Path to a configuration file")
+			man := NewManager(&cmd)
+
+			// load the yaml config
+			man.viper.SetConfigType("yaml")
+			require.NoError(t, man.viper.ReadConfig(strings.NewReader(c.yaml)))
+
+			// TODO: tried to test command-line flags too by using cmd.SetArgs to
+			// test-case values, but that didn't seem to work, not sure how it can
+			// be done in our particlar setup.
```

Comment:
Haven't found a way to do this but I've tested the flags manually. Would be a nice-to-have if we find a way to support this eventually.